pkg/virtual/framework/forwardingregistry/store.go

```diff
@@ -262,14 +265,23 @@ func clientGetter(dynamicClusterClient dynamic.ClusterInterface, namespaceScoped
 			}
 		}
 
-		if namespaceScoped {
-			if namespace, ok := genericapirequest.NamespaceFrom(ctx); ok {
-				return dynamicClusterClient.Cluster(clusterName).Resource(gvr).Namespace(namespace), nil
-			} else {
-				return nil, fmt.Errorf("there should be a Namespace context in a request for a namespaced resource: %s", gvr.String())
-			}
-		} else {
-			return dynamicClusterClient.Cluster(clusterName).Resource(gvr), nil
+		if !namespaceScoped {
+			return &transforming.TransformingClient{
+				Transformations: transformers,
+				Client:          dynamicClusterClient.Cluster(clusterName).Resource(gvr),
+				Namespace:       "",
+				Resource:        gvr,
+			}, nil
+		}
+
+		if namespace, exists := genericapirequest.NamespaceFrom(ctx); exists {
+			return &transforming.TransformingClient{
+				Transformations: transformers,
+				Client:          dynamicClusterClient.Cluster(clusterName).Resource(gvr).Namespace(namespace),
+				Namespace:       namespace,
+				Resource:        gvr,
+			}, nil
```

Comment:
this structure looks like it could be a wrapper of a client? Why inject it deep into the clientGetter?

Reply:
Do the changes done in commit 3f0d7ee5e0878187dce7f08113d72eed48d8ba54 correspobd to what you had in mind ?